src/files/manager.ts

```diff
@@ -0,0 +1,153 @@
+   * ```typescript
+   * const filesManager = new FilesManager(httpClient);
+   *
+   * const files = await filesManager.find({
+   *   filters: { mime: { $contains: 'image' } },
+   *   sort: 'name:asc',
+   * });
+   *
+   * console.log(files); // Array of file objects
+   * ```
+   */
+  async find(queryParams?: FileQueryParams): Promise<FileListResponse> {
+    debug('finding files');
+
+    try {
+      // Validate query parameters before making the request
+      if (queryParams) {
+        validateFileQueryParams(queryParams);
+      }
+
+      let url = '/upload/files';
+
+      if (queryParams) {
+        url = URLHelper.appendQueryParams(url, queryParams);
+      }
+
+      const response = await this._httpClient.get(url);
+      const json = await response.json();
+
+      debug('found %o files', Number(json?.length));
+
+      return json;
+    } catch (error) {
+      // Handle validation errors which are already Error instances
+      if (error instanceof Error) {
+        debug('error finding files: %o', error.message);
+        throw error;
+      }
+
+      // Pass through HTTP errors from the HttpClient
+      debug('unexpected error finding files: %o', error);
+      throw error;
+    }
+  }
+
+  /**
+   * Retrieves a single file by its ID.
+   *
+   * @param fileId - The numeric identifier of the file to retrieve.
+   * @returns A promise that resolves to a single file object.
+   *
+   * @throws {HTTPError} if the HTTP client encounters connection issues, the server is unreachable, or authentication fails.
+   * @throws {Error} if the file ID is invalid or the file does not exist.
+   *
+   * @example
+   * ```typescript
+   * const filesManager = new FilesManager(httpClient);
+   *
+   * const file = await filesManager.findOne(1);
+   *
+   * console.log(file); // File object with details
+   * ```
+   */
+  async findOne(fileId: number): Promise<FileResponse> {
+    validateFileId(fileId);
+
+    debug('finding file with ID %o', fileId);
+
+    const url = `/upload/files/${fileId}`;
+
+    try {
+      const response = await this._httpClient.get(url);
+      const json = await response.json();
+
+      // Check if the response is empty or doesn't have expected structure
+      if (!json || typeof json !== 'object') {
+        throw new Error(
+          `File with ID ${fileId} not found or returned invalid data. Expected a file object but received ${typeof json}.`
+        );
+      }
+
+      debug('found file with ID %o', fileId);
+
+      return json;
+    } catch (error) {
+      debug('error finding file with ID %o: %o', fileId, error);
+
+      // Handle 404 errors with a more specific message
+      if ((error as Response)?.status === 404) {
+        throw new Error(
+          `File with ID ${fileId} not found. The requested file may have been deleted or never existed.`
+        );
+      }
+
+      // For other HTTP errors, provide context about the operation
+      if ((error as Response)?.status) {
```

Comment:
I think you can check using `instanceof HTTPError`?

See `src/interceptors/auth.ts:120` for more info (same for above)